haystack/schema.py

```diff
@@ -41,12 +34,13 @@
 
 @dataclass
 class Document:
-    content: Union[str, pd.DataFrame]
-    content_type: Literal["text", "table", "image", "audio"]
     id: str
-    meta: Dict[str, Any]
+    content: Union[str, pd.DataFrame]
+    content_type: Literal["text", "table", "image", "audio"] = Field(default="text")
+    meta: Dict[str, Any] = Field(default={})
```

Comment:
A dataclass is not constrained by its initialization, but how, in the essence, it threatens each own field.

Looking over the meta field in the whole code, I found that in every DocumentStore it's handled as being existent. If you try to force it as None, you will have to fix code over all written document stores. Take a look. It's never handled the None scenario. It could even someday, if someone removed the __init__ assignment, the error be propagated to the entire codebase.

Then I reviewed the primitive Document. There you will see, in the past, a custom initialization to `meta = meta or { } `.

So, the __init__ was indeed checking if it was None and then forcing an empty dict. What do we learn from this?

meta should never be None. It's a fact the primitive class designer has chosen.

But, to help users with no need to send extra parameters, which could be rarely used, he made it Optional. I removed that from the dataclass.

I have left the original declaration pattern just to "keep the pattern". And remembering that the init parameters doesn't need to be the same as the fields dataclass. They are, in the ideal way, not to be used. But occasionally, they need to.

In the essence, I think that meta should not be Optional, neither on a custom initialization. Why? Because you may send the user of the class the wrong impression that a value of None is ok to be sent (Optional is Union[ ,None]), which in the essence is not, as we are always initializing to the empty Dict.